packages/permission-log-controller/src/PermissionLogController.ts

```diff
@@ -333,65 +277,57 @@ export class PermissionLogController extends BaseController<
    * @param time - The time of the request, i.e. Date.now().
    * @param isEthRequestAccounts - Whether the permissions request was 'eth_requestAccounts'.
    */
-  logPermissionsHistory(
+  #logPermissionsHistory(
     requestedMethods: string[],
     origin: string,
     result: Json,
     time: number,
     isEthRequestAccounts: boolean,
   ) {
-    let accounts: string[];
     let newEntries: PermissionEntry;
 
     if (isEthRequestAccounts) {
       // Type assertion: We are assuming that the response data contains
       // a set of accounts if the RPC method is "eth_requestAccounts".
-      accounts = result as string[];
-      const accountToTimeMap = getAccountToTimeMap(accounts, time);
-
+      const accounts = result as string[];
       newEntries = {
         eth_accounts: {
-          accounts: accountToTimeMap,
+          accounts: getAccountToTimeMap(accounts, time),
           lastApproved: time,
         },
       };
     } else {
       // Records new "lastApproved" times for the granted permissions, if any.
       // Special handling for eth_accounts, in order to record the time the
       // accounts were last seen or approved by the origin.
-
       // Type assertion: We are assuming that the response data contains
       // a set of permissions if the RPC method is "eth_requestPermissions".
-      newEntries = (result as Permission[])
-        .map((perm) => {
-          if (perm.parentCapability === 'eth_accounts') {
-            accounts = this.getAccountsFromPermission(perm);
```

Comment:
Nice refactor here — it actually addresses a possible bug where if multiple permission objects have a `parentCapability`, `accounts` would only reflect the last one.